src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -1699,6 +1699,31 @@ public IReadOnlyList<SyntaxNode> GetStatementContainerStatements(SyntaxNode node
                ? GetExecutableBlockStatements(node)
                : (IReadOnlyList<SyntaxNode>)ImmutableArray.Create<SyntaxNode>(node.GetEmbeddedStatement());
 
+        public bool IsTerminator(SyntaxNode node, SyntaxToken token)
```

Comment:
ok, based on the docs in ISyntaxFacts, isn't this just seeing if this token is the same as the node's ChildNOdesOrTokens.Last()?  if not, i'm curiosu about which case it isn't.

Reply:
this comment still applies.

Reply:
It does not apply for the following cases:

* The last token is missing
* The last token is part of a wildcard
* The last token is followed by an optional token

Reply:
> The last token is missing

The code below doesn't seem to handle that either.

> The last token is part of a wildcard (e.g. variable_declaration)

It's unclear to me what *would* be the terminator in that case.  

> The last token is followed by an optional token

Same issue.  What is the 'terminator' for a construct that takes optional constraints?

Reply:
honestly, this helper is called in one place:

```c#
leftToken.GetAncestor(syntaxFacts.IsUsingOrExternOrImport) is { } node
                && !syntaxFacts.IsTerminator(node, leftToken);
```

And all it's asking is "are we after the semicolon of that statement?"  I honestly don't see any need for the helper.  We can just inline this to teh caller's class to do this check.

Reply:
> It's unclear to me what would be the terminator in that case.

There is no terminator for this case. There is no way to look at a single token and know that it will be the definitive final token of the syntax node.

>  We can just inline this to teh caller's class to do this check.

The caller is language agnostic. Conceptually this is a broader concept though: it answers the question of whether the next typed token could be part of the same syntax node as the previously-typed token.

Reply:
> The caller is language agnostic. 

but in an abstract class.  i'd have them just ask that question in an abstract methot that the derived types fulfill.

> Conceptually this is a broader concept though: it answers the question of whether the next typed token could be part of the same syntax node as the previously-typed token.

i can see that concept.  but we're simply not utilizing it in that way, and i would really prefer this not to go into syntaxfacts without a strong belief taht this is needed for many clients.

Reply:
I removed IsTerminator and added `IsFinalSemicolonOfUsingOrExtern` for just this feature